op-node/p2p/sync.go

```diff
@@ -556,6 +564,13 @@ func (s *SyncClient) peerLoop(ctx context.Context, id peer.ID) {
 	// so we don't be too aggressive to the server.
 	rl := rate.NewLimiter(peerServerBlocksRateLimit, peerServerBlocksBurst)
 
+	// if onlyReqToStatic is on, ensure that only static peers are dealing with the request
+	peerRequests := s.peerRequests
+	if s.syncOnlyReqToStatic && !s.extra.IsStatic(id) {
+		// for non-static peers, set peerRequests to nil
```

Comment:
Please clarify in the comment that this will effectively make the peer loop not perform outgoing sync-requests. And add a note that while sync-requests will block, the loop may still process other events (if added in the future).